.github/workflows/build-push.yml

```diff
@@ -0,0 +1,289 @@
+          
+            if [ -z "$LATEST_RELEASE" ]; then
+              echo "No releases found, using default version 0.0.0"
+              LATEST_RELEASE="0.0.0"
+            else
+              echo "Latest release found: $LATEST_RELEASE"
+            fi
+          
+            # Normalize branch name for semver pre-release identifier
+            # Replace invalid characters with hyphens and convert to lowercase
+            NORMALIZED_BRANCH=$(echo "${{ github.ref_name }}" | sed 's/[^a-zA-Z0-9-]/-/g' | sed 's/--*/-/g' | sed 's/^-\|-$//g' | tr '[:upper:]' '[:lower:]')
+          
+            # Construct semver with pre-release identifier
+            PACKAGE_VERSION="${LATEST_RELEASE}-${NORMALIZED_BRANCH}"
+            echo "Generated package version: $PACKAGE_VERSION"
+          fi
+          
+          echo "PACKAGE_VERSION=$PACKAGE_VERSION" >> $GITHUB_ENV
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_USER_SSH_KEY && secrets.GITHUB_TOKEN || github.token }}
+
+      - name: Inspect origin branch and determine build branch
+        run: |
+          ORIGIN_BRANCH="${{ github.ref_name }}"
+          echo "Origin branch: $ORIGIN_BRANCH"
+          
+          # Determine the build branch name
+          if [ -n "${{ inputs.BUILT_BRANCH_NAME }}" ]; then
+            BUILD_BRANCH="${{ inputs.BUILT_BRANCH_NAME }}"
+          else
+            BUILD_BRANCH="dev/$ORIGIN_BRANCH"
+          fi
+          
+          echo "Build branch: $BUILD_BRANCH"
+          echo "BUILT_BRANCH_NAME=$BUILD_BRANCH" >> $GITHUB_ENV
+
+      - name: Checkout the build branch and merge in it
+        run: |
+          git show-ref -q refs/remotes/origin/${{ env.BUILT_BRANCH_NAME }} && git checkout ${{ env.BUILT_BRANCH_NAME }} || git checkout -b ${{ env.BUILT_BRANCH_NAME }}
+          git checkout ${{ github.ref_name }} -- .
+
+      - name: Set up PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ inputs.PHP_VERSION }}
+          tools: php-scoper, sniccowp/php-scoper-wordpress-excludes, composer/installers, inpsyde/wp-translation-downloader, ${{ inputs.PHP_TOOLS }}
+          coverage: none
+
+      - name: Install Composer dependencies without dev dependencies
+        uses: ramsey/composer-install@v3
+        env:
+          COMPOSER_AUTH: '${{ secrets.COMPOSER_AUTH_JSON }}'
+        with:
+          composer-options: ${{ inputs.COMPOSER_ARGS }}
+
+      - name: Set up node
+        uses: actions/setup-node@v4
+        env:
+          NODE_OPTIONS: ${{ inputs.NODE_OPTIONS }}
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_REGISTRY_TOKEN }}
+        with:
+          node-version: ${{ inputs.NODE_VERSION }}
+          registry-url: ${{ inputs.NPM_REGISTRY_DOMAIN }}
+          cache: npm
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Set plugin version header
+        if: ${{ hashFiles('style.css') == '' }}
```

Comment:
`style.css` is part of WordPress Themes. Do we need to adjust the `if:`-condition here to `index.php` or do we need to update the `name:` to "Theme"? 👀

Reply:
Yeah we've been tripping over this before. I think we can afford a variable that clearly tells us what this is about